src/core/lib/event_engine/cf_engine/cf_engine.cc

```diff
@@ -0,0 +1,140 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <grpc/support/port_platform.h>
+
+#ifdef GPR_APPLE
+
+#include "src/core/lib/event_engine/cf_engine/cf_engine.h"
+#include "src/core/lib/event_engine/posix_engine/timer_manager.h"
+#include "src/core/lib/event_engine/trace.h"
+#include "src/core/lib/event_engine/utils.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+
+class CFEventEngine::Closure final : public EventEngine::Closure {
```

Comment:
nit: per the C++ style guide, you should either make this a `struct`, or a class with private members and appropriate setters/getters. See https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes. I'd suggest keeping it a class, given this excerpt from the style guide:

> The struct must not have invariants that imply relationships between different fields, since direct user access to those fields may break those invariants. 

Something like this could work:
```
class CFEventEngineClosure final : public EventEngine::Closure {
 public:
  CFEventEngineClosure(absl::AnyInvocable<void()> cb, EventEngine* engine)
      : cb_(std::move(cb)), engine_(engine) {}

  const EventEngine::TaskHandle& handle() const { return handle_; }
  void SetHandle(EventEngine::TaskHandle handle) { handle_ = handle; }

  void Run() override {
    GRPC_EVENT_ENGINE_TRACE("CFEventEngine:%p executing callback:%s", engine_,
                            HandleToString(handle_).c_str());
    {
      grpc_core::MutexLock lock(&engine_->mu_);
      engine_->known_handles_.erase(handle_);
    }
    cb_();
    delete this;
  }

 private:
  absl::AnyInvocable<void()> cb_;
  CFEventEngine* engine_;
  EventEngine::TaskHandle handle_;
};
```

Reply:
Will change it to struct as fields are used by EventEngine class. This is also how posix engine and windows engine do.

Reply:
That's fine. I'm not arguing that we did the others correctly with respect to the style guide ... I think maybe we didn't, but it's rather minor.

Reply:
@drfloob Hey AJ would you mind LGTM too if this looks OK for you? thanks!

Reply:
LGTM!